### ISSUE TITLE
Multiple values or not?

### ISSUE THREAD

**Issue by robert-kisteleki** [mahtin/http-source-information-header#4]
The ultimate goal is to add a signal to the client about "what server answered this particular query?" However, the answer itself might have been assembled by querying (multiple) upstreams, all of which could have added their server-id. 

Should we allow this, and end up with something like
`
Server-ID: whatever; whatever else; something else
`

This complicates things. It also seems to be a very specific corner case. However, I do see the potential value for CDNs, as long as they are ok with exposing these bits of their internal workings.

And if we add it: should "the server answered this particular query?" be the first or the last in the list?

**Comment by cbranch** [comment]
There is precedent for this in the Via header; I suggest matching its behaviour.

Similarly, the delimiter in the text should probably be a comma and not a semicolon, to match the guidance in RFC 7230 Section 3.2.2.

**Comment by robert-kisteleki** [comment]
The question still stands: is there a point for multiple values _for this header, this purpose_?

**Comment by mahtin** [comment]
Yes. There is a valid reason for multiple values being returned in the Server-ID header, as long at the order is well defined in the RFC draft. Assume that each step is additive to the end of the lines, then if an end-user only wants the final server information, they can extract it via javascript.

``` javascript
  // modified from ... https://www.edge-cloud.net/2015/10/04/cloudflare-pops-in-google-analytics/
  var req = new XMLHttpRequest();
  req.open('HEAD', document.location, false);
  req.send(null);
  var server_id = req.getResponseHeader("Server-ID").split(","); // trivial comma processing for now
  var last_server_id = server_id[server_id.length - 1];
  console.log(last_server_id);
```

If they want the whole path (assuming that servers respond appropriately); then the data is still there.

**Comment by robert-kisteleki** [comment]
I can live with any of the orderings, but indeed it must be well defined
